File: app/src/main/java/mchehab/com/mnist/MainActivity.java
package mchehab.com.mnist;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import android.widget.Button;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    private TextView textViewResult;
    private DrawView drawView;
    private Button buttonClear;
    private Button buttonPredict;

    private TFClassifier tfClassifier;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        tfClassifier = new TFClassifier(getApplicationContext());

        textViewResult = findViewById(R.id.textViewResult);
        drawView = findViewById(R.id.drawView);
        buttonClear = findViewById(R.id.buttonClear);
        buttonPredict = findViewById(R.id.buttonPredict);

        buttonClear.setOnClickListener(e -> drawView.clearCanvas());

        buttonPredict.setOnClickListener(e -> {
            float[] result = tfClassifier.predict(drawView.getPixels());
            float max = -1;
            int answer = -1;
            for(int i=0;i<result.length;i++){
                if(result[i] > max){
                    max = result[i];
                    answer = i;
                }
            }
            textViewResult.setText("Model Prediction: " + answer + "\nConfidence: " + max);
        });
    }
}

File: kotlin/src/main/java/mchehab/com/kotlin/MainActivity.kt
package mchehab.com.kotlin

import android.support.v7.app.AppCompatActivity
import android.os.Bundle

import android.widget.Button
import android.widget.TextView

class MainActivity : AppCompatActivity() {

    private var textViewResult: TextView? = null
    private var drawView: DrawView? = null
    private var buttonClear: Button? = null
    private var buttonPredict: Button? = null

    private var tfClassifier: TFClassifier? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        tfClassifier = TFClassifier(applicationContext)

        textViewResult = findViewById(R.id.textViewResult)
        drawView = findViewById(R.id.drawView)
        buttonClear = findViewById(R.id.buttonClear)
        buttonPredict = findViewById(R.id.buttonPredict)

        buttonClear!!.setOnClickListener { e -> drawView!!.clearCanvas() }

        buttonPredict!!.setOnClickListener { e ->
            val result = tfClassifier!!.predict(drawView!!.pixels)
            var max = -1f
            var answer = -1
            for (i in result.indices) {
                if (result[i] > max) {
                    max = result[i]
                    answer = i
                }
            }
            textViewResult!!.text = "Model Prediction: $answer\nConfidence: $max"
        }
    }
}